src/psyclone/tests/psyGen_test.py

```diff
@@ -3167,6 +3178,51 @@ def test_symbol_initialisation():
     assert ("Symbols that are part of another symbol shape can "
             "only be scalar integers, but found") in str(error.value)
 
+    with pytest.raises(ValueError) as error:
+        Symbol('a', 'integer', scope='global_argument', constant_value=9)
+    assert ("Symbol with a constant value is currently limited to having "
+            "local scope but found 'global_argument'.") in str(error)
+
+    with pytest.raises(ValueError) as error:
+        Symbol('a', 'integer', shape=[None], constant_value=9)
+    assert ("Symbol with a constant value must be a scalar but the shape "
+            "attribute is not empty.") in str(error)
+
+    with pytest.raises(ValueError) as error:
+        Symbol('a', 'integer', constant_value=9.81)
+    assert ("This Symbol instance's datatype is 'integer' which means the "
+            "constant value is expected to be") in str(error)
+    assert "'int'>' but found " in str(error)
+    assert "'float'>'." in str(error)
+
+    with pytest.raises(ValueError) as error:
+        Symbol('a', 'character', constant_value=42)
+    assert ("This Symbol instance's datatype is 'character' which means the "
+            "constant value is expected to be") in str(error)
+    assert "'str'>' but found " in str(error)
+    assert "'int'>'." in str(error)
+
+    with pytest.raises(ValueError) as error:
+        Symbol('a', 'boolean', constant_value="hello")
+    assert ("This Symbol instance's datatype is 'boolean' which means the "
+            "constant value is expected to be") in str(error)
+    assert "'bool'>' but found " in str(error)
+    assert "'str'>'." in str(error)
+
+
+def test_symbol_map():
+    '''Test the mapping variable in the Symbol class does not raise any
+    exceptions when it is used with the valid_data_types variable in
+    the Symbol class.
+
+    '''
+    # "real" is not supported in the mapping so we expect it to have 1
+    # less entry than there are valid data types.
+    assert len(Symbol.valid_data_types) == len(Symbol.mapping) + 1
+    for data_type in Symbol.valid_data_types:
+        if data_type not in ["real"]:
+            Symbol.mapping[data_type]
```

Comment:
Pylint says "pointless-statement: Statement seems to have no effect", I guess you are just checking the map contains the datatype, would `assert data_type in Symbo.mapping` be equivalent?

Reply:
Yes, that's a better way to do it.